### PR TITLE
trying assignment 4

### PR DESCRIPTION
**Pull request by mikejerome**
I was too busy in the last week and ran out of time for this one :cry:

### CODE REVIEW

File: lib/private.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        server.auth.strategy('simple', 'basic', { validateFunc: internals.validate });
+        server.route({
+            method: 'GET',
+            path: '/private',
+            config: {
+                description: 'Returns welcome page protected by basic auth',
+                auth: 'simple',
+                handler: function (request, reply) {
+
+                    return reply('hello');
```
Comment (AdriVanHoudt) [inline lib/private.js]: the reply should at least have the (user)name of the logged in user in it

File: package.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
   "homepage": "https://github.com/hueniverse/hueniversity",
   "dependencies": {
     "hapi": "8.x.x",
+    "hapi-auth-basic": "^2.0.0",
```
Comment (AdriVanHoudt) [inline package.json]: try using the 2.x.x style notation

File: test/private.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                expect(res.statusCode, 'Status code').to.equal(401);
+
+                server.stop(done);
+            });
+        });
+    });
+
+    it('returns error on wrong username', function (done) {
+
+        Hueniversity.init(0, function (err) {
```
Comment (idanwe) [inline test/private.js]: you are missing the server argument `(err, server)`

**Comment by mikejerome** [comment]
Thanks a lot for the help @idanwe and @AdriVanHoudt

**Pull request closed without merging**